cli/lsp/resolver.rs

```diff
@@ -218,7 +219,53 @@ impl LspScopeResolver {
               continue;
             };
             let specifier = into_specifier_and_media_type(Some(url)).0;
-            deps.push((specifier, name.clone()))
+            if dep_package_json.is_none() {
+              dep_package_json = (|| {
+                let path = url_to_file_path(&specifier).ok()?;
+                pkg_json_resolver.get_closest_package_json(&path).ok()?
+              })();
+            }
+            deps.push((specifier, name.clone()));
+          }
+          let export_keys = dep_package_json
+            .as_ref()
+            .and_then(|p| p.exports.as_ref())
+            .into_iter()
+            .flat_map(|e| e.keys());
+          for export_key in export_keys {
+            let Some(export_suffix) = export_key.strip_prefix("./") else {
+              continue;
+            };
+            let name_and_export = format!("{name}/{export_suffix}");
```

Comment:
What happens if it contains a wildcard?

Reply:
I might have to scan directories to support them, or skip them for now

Reply:
There's some code that does it in https://github.com/denoland/deno/pull/28131/files#diff-9cf4c6f84419e1517fd293cc725422ed4b17ce6c596fd0dc07bc6c1c95a6a671R1859 (I'm hesitant to keep working on that PR further atm, but it would make it work better with typescript).

Reply:
I added a lookup cache in the node resolver, wildcard cases would fail to resolve here and use that logic instead.